apps/src/code-studio/components/progress/lessonLockDialog/LessonLockDialog.jsx

```diff
@@ -77,10 +78,19 @@ function LessonLockDialog({
   const handleSave = async () => {
     setSaving(true);
     const csrfToken = $('meta[name="csrf-token"]').attr('content');
-    await saveLockState(serverLockState, clientLockState, csrfToken);
-    // Refresh the lock information on the script overview page and teacher panel
-    await refetchSectionLockStatus(selectedSectionId, unitId);
-    handleClose();
+    const saveLockStateResponse = await saveLockState(
+      serverLockState,
+      clientLockState,
+      csrfToken
+    );
+    if (saveLockStateResponse.ok) {
+      // Refresh the lock information on the script overview page and teacher panel
+      await refetchSectionLockStatus(selectedSectionId, unitId);
+      handleClose();
+    } else {
+      setSaving(false);
+      setError(commonMsg.errorSavingLockStatus());
```

Comment:
One scenario is in my mind here: Suppose a user hits Save, it doesn't work, they get the error message, and then they hit Save again. One of two things happen:
1) It doesn't work again –– the message is still there, and the user sees it still didn't work.
2) It does work this time –– the dialogue closes, so they assume it worked, which is did.
^^ Is that right? I'm trying to make sure the error message doesn't persist if it works the second time. It looks like it won't because the dialogue disappears?

Reply:
Yes that's correct! We don't show an explicit success message when the request succeeds. I could clear the error message when the user hits save so that, when it appears again, it's obvious that the request is still failing

Reply:
With the error message cleared out, here's what those two cases look like.

Error then error:
https://user-images.githubusercontent.com/46464143/181824088-b8fbdb42-f1e9-4628-90bb-c8753dae1a75.mov

Error then success:
https://user-images.githubusercontent.com/46464143/181824098-768e7065-8779-440b-8f5f-420467670814.mov

Reply:
That looks great to me!